ibc-clients/ics07-tendermint/types/src/client_state.rs

```diff
@@ -364,223 +364,3 @@ impl From<ClientState> for Any {
         }
     }
 }
-
-#[cfg(all(test, feature = "serde"))]
-pub(crate) mod serde_tests {
-    use serde::de::DeserializeOwned;
-    use serde::Serialize;
-    use tendermint_rpc::endpoint::abci_query::AbciQuery;
-
-    pub fn test_serialization_roundtrip<T>(json_data: &str)
-    where
-        T: core::fmt::Debug + PartialEq + Serialize + DeserializeOwned,
-    {
-        let parsed0 = serde_json::from_str::<T>(json_data);
-        assert!(parsed0.is_ok());
-        let parsed0 = parsed0.unwrap();
-
-        let serialized = serde_json::to_string(&parsed0);
-        assert!(serialized.is_ok());
-        let serialized = serialized.unwrap();
-
-        let parsed1 = serde_json::from_str::<T>(&serialized);
-        assert!(parsed1.is_ok());
-        let parsed1 = parsed1.unwrap();
-
-        assert_eq!(parsed0, parsed1);
-    }
-
-    #[test]
-    fn serialization_roundtrip_no_proof() {
-        let json_data = include_str!(concat!(
-            env!("CARGO_MANIFEST_DIR"),
-            "/../../../ibc-testkit/tests/data/json/client_state.json"
-        ));
-        test_serialization_roundtrip::<AbciQuery>(json_data);
-    }
-
-    #[test]
-    fn serialization_roundtrip_with_proof() {
-        let json_data = include_str!(concat!(
-            env!("CARGO_MANIFEST_DIR"),
-            "/../../../ibc-testkit/tests/data/json/client_state_proof.json"
-        ));
-        test_serialization_roundtrip::<AbciQuery>(json_data);
-    }
-}
-
-#[cfg(test)]
-mod tests {
```

Comment:
I think, we can leave these tests here. IMO these are just unit tests.

Reply:
[4fe1394](https://github.com/cosmos/ibc-rs/pull/1233/commits/4fe139424c90f1113f4004dc00176e3b86d0d90e)